version.go

```diff
@@ -52,6 +62,25 @@ func Version() string {
 	return version
 }
 
+// UserAgent returns the full user agent string that identifies the software
+// that is submitting swaps to the loop server.
+func UserAgent() string {
+	// Start with the major, minor, and patch versions.
+	version := fmt.Sprintf("%d.%d.%d", appMajor, appMinor, appPatch)
+
+	// Append pre-release version if there is one.  The hyphen called for
+	// by the semantic versioning spec is automatically appended and should
+	// not be contained in the pre-release string.  The pre-release version
+	// is not appended if it contains invalid characters.
+	preRelease := normalizeVerString(appPreRelease)
```

Comment:
nit: put this in a versionStr() function rather than duplicate what we already have in `Version`?